packages/services/api/src/modules/shared/__tests__/mutex.spec.ts

```diff
@@ -31,43 +31,6 @@ it('should allow different locks at any time', async ({ expect }) => {
   await expect(mutex.lock('3', { signal })).resolves.toBeTruthy();
 });
 
-it('should cancel locking on abort signal', async ({ expect }) => {
-  const mutex = new Mutex(new Tlogger(), new Redis(differentPort()));
-
-  const [signal, abort] = createSignal();
-
-  const unlock1 = await mutex.lock('1', { signal });
-
-  const lock2 = mutex.lock('1', { signal });
-
-  abort();
-
-  await expect(lock2).rejects.toMatchInlineSnapshot('[Error: Locking aborted]');
-
-  unlock1();
-
-  // make sure that the aborted lock does not lock
-  await expect(mutex.lock('1', { signal: createSignal()[0] })).resolves.toBeTruthy();
-});
-
-it('should unlock on abort signal', async ({ expect }) => {
-  const mutex = new Mutex(new Tlogger(), new Redis(differentPort()));
-
-  const [signal, abort] = createSignal();
-
-  await mutex.lock('1', { signal });
-
-  const lock2 = mutex.lock('1', { signal: createSignal()[0] });
-
-  // second lock shouldnt resolve
-  await expect(Promise.race([throwAfter(50), lock2])).rejects.toBeTruthy();
-
-  abort();
-
-  // first lock is aborted, second one should resolve now
-  await expect(lock2).resolves.toBeTruthy();
-});
```

Comment:
if the abort signal/request is canceled, but the action is already being performed, we don't want to release the lock. Instead, we wait until the action is performed or the global resource lock timeout is exceeded.